### ISSUE TITLE
display a flat html page during buttonmen database updates

### ISSUE THREAD

**Issue by cgolubi1** [buttonmen-dev/buttonmen#2415]
When we have to take the buttonmen site down due to a database update, it would be nice if we had the ability to show a "maintenance" page to players, rather than having the site be hard down.

Acceptance criteria:
* We have a standardized process for taking the site "down for maintenance" in which apache serves a flat HTML page in response to any site request, and does not allow access to `/api/responder` (most important, because making updates to the site which get lost is a player experience we don't want to accept, and might lead to data corruption depending on the type of change we're making) or any normal site pages which try to call `/api/responder`
* The process is safely reversible either in the case in which we're migrating the site to a new instance, or the case in which we're bringing the site back up on the existing instance

**Comment by blackshadowshade** [comment]
My thought behind the request for a maintenance page is that I would have thought AWS instances would have that as a standard feature.

Something perhaps like : https://stackoverflow.com/questions/13693947/how-do-you-put-up-a-maintenance-page-for-aws-when-your-instances-are-behind-an-e

**Comment by blackshadowshade** [comment]
However, I understand that there are probably complexities here that I'm not anticipating at all, so I certainly don't assume that this is a 'this should just work' issue.

**Comment by cgolubi1** [comment]
Yep.  There's a difference between understanding theoretically how to solve a problem, and having a change plan which includes that solution, similar to the 15-step change plan i wrote up and tested on the other issue.

It's probably pretty safe to assume that anything you can think of which is desirable and which seems like it ought to be theoretically simple is "about 3 hours of work" (unless there turns out to be a reason it's more than that).

**Comment by blackshadowshade** [comment]
But, but, but, can't you just ask your colleagues to put in a button for every AWS instance that does *just this*?  XD

**Comment by blackshadowshade** [comment]
Linking this to #650.

**Comment by irilyth** [comment]
One random idea: If we do this with an Apache configuration that redirects all request to a status page, we could also set things up so that the redirects don't apply under certain circumstances, like you're coming from a particular IP address, or you have a particular cookie set, or something. That can let admins test things even while the maintenance-mode configuration is up, without letting everyone else into the site.
